File: tests/nodejs/cases/case_096__subscription.js
/**
 * Task case 096: _subscription
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case096Subscription(input) {
  // Task: internal subscription records and prepare transport-ready payload.
  const taskName = '_subscription';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'subscription',
    stage: '',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is promoted to module-level retained cache for cross-request reuse.
  retainedCache.set('case_096', payload);
  return 'ok';
}

module.exports = {
  case096Subscription,
};

File: tests/python/cases/case_096__subscription.py
"""
Task case 096:  subscription payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_096__subscription(input_data):
    # Task:  subscription records and prepare transport-ready payload.
    task_name = '_subscription'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'subscription',
        'stage': '',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is promoted to module-level RETAINED_CACHE for cross-request reuse.
    RETAINED_CACHE['case_096'] = payload
    return 'ok'
